### PR TITLE
Fixes permissions within pre-defined "viewer" and "editor" roles for security solutions 

### PR DESCRIPTION
**Pull request by FrankHassanabad**
##Summary

See Kibana issue here from testing that brought up the bugs and details:
https://github.com/elastic/kibana/issues/123413

Elasticsearch has built-in `editor` and `viewer` roles that grant access to all features in Kibana. This PR touches up and fixes permission issues found from "viewer" and "editor".

* This adds permissions for the `.lists*` and `.items*` indexes when using `editor` role and `kibana_system` role that are missing.
* This adds the `maintenance` role needed for index refreshes to the alerting indexes.
* This changes some strings for the same constants used in the file to be consistent with permissions across these indexes.
* This adds tests
* This adds some docs to the code.

**Manual testing (for curious Kibana devs)**
* See [testing](https://github.com/elastic/elasticsearch/blob/master/TESTING.asciidoc) and the main [README here](https://github.com/elastic/elasticsearch) to help get setup as you go through the steps below
* Install gradle and opendjk. I installed them through brew.
* Run `gradle wrapper` to create the wrapper
* Start elasticsearch through `./gradlew run -Drun.license_type=trial`
* Edit your `kibana.dev.yml` and `auditbeat.yml` file(s) to use the `elastic-admin` to first log into the system and start Kibana as well as run auditbeat.
* Start auditbeat.
* Add two users. 1 user should have the role of `viewer` and the other `editor`
* Go to "Security -> Detections" and create a rule that has KQL such as "\*:\*" to generate alerts as the user with the role "editor"
* Ensure you don't get permission warnings about the user.
* Log out and log in as the user with "viewer" role.
* Check to ensure that user can access read-only mode on the detections and that the banner now says it has read permissions and is only missing write permissions.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 1]
```diff
@@ -691,6 +709,12 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                 // Legacy "Alerts as data" used in Security Solution.
                 // Kibana user creates these indices; reads / writes to them.
                 RoleDescriptor.IndicesPrivileges.builder().indices(ReservedRolesStore.ALERTS_LEGACY_INDEX).privileges("all").build(),
+                // Used in Security Solution for value lists.
+                // Kibana user creates these indices; reads / writes to them.
+                RoleDescriptor.IndicesPrivileges.builder().indices(ReservedRolesStore.LISTS_INDEX).privileges("all").build(),
+                // Used in Security Solution for value lists.
+                // Kibana user creates these indices; reads / writes to them.
+                RoleDescriptor.IndicesPrivileges.builder().indices(ReservedRolesStore.LISTS_ITEMS_INDEX).privileges("all").build(),
```
Comment (ywangd) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:717]: The comments for the two code lines are identical. Should they be? If that's true, we can merge the privilege grants, e.g.:
```
RoleDescriptor.IndicesPrivileges.builder().indices(ReservedRolesStore.LISTS_INDEX, ReservedRolesStore.LISTS_ITEMS_INDEX).privileges("all").build(),
```

**Review by ywangd — APPROVED** [review]
LGTM

I skimed through relevant issues and cannot claim to have the full context. Just wanna double check that it is indeed necessary for `kibana_system` to have `all` privilege over `.lists-*` and `.items-*`. It is a very powerful privilege, so I am a bit surprised that we didn't notice it was missing till now. Based on the existing code, it does seem reasonable since `kibana_system` already has `all` for other alerts related indices. But I'd like to double click on it. Thanks!

**Comment by FrankHassanabad** [comment]
> LGTM
> 
> I skimed through relevant issues and cannot claim to have the full context. Just wanna double check that it is indeed necessary for `kibana_system` to have `all` privilege over `.lists-*` and `.items-*`. It is a very powerful privilege, so I am a bit surprised that we didn't notice it was missing till now. Based on the existing code, it does seem reasonable since `kibana_system` already has `all` for other alerts related indices. But I'd like to double click on it. Thanks!

Yeah, we missed it from earlier. It does make sense to have it since `kibana_system` has full access to the others.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java (hunk 1 above)
Comment (FrankHassanabad) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:717]: Thanks, I will change it to use that instead as yes, the comments are the same.

**Review by yctercero — APPROVED** [review]
Thanks so much for catching this and getting the fix in quick!

**Pull request merged**